tensorflow/core/grappler/optimizers/auto_mixed_precision.cc

```diff
@@ -1990,10 +1994,16 @@ int GetNumGPUs(const Cluster& cluster,
   int num_gpus = 0;
   for (const auto& device : devices) {
     const DeviceProperties& device_properties = device.second;
+#if GOOGLE_CUDA
     std::pair<int, int> arch = GetDeviceGPUArch(device_properties);
     if (device_properties.type() == "GPU" && arch >= min_arch) {
       num_gpus++;
     }
+#elif TENSORFLOW_USE_ROCM
+    if (device_properties.type() == "GPU" && GetFastFP16Support(device_properties)){
+         num_gpus++; 
+    }
+#endif
```

Comment:
@stevenireeves 
this logic is flawed in the case where `ShouldIgnorePerformance()` returns true
 (which will be the case for the `//tensorflow/python:auto_mixed_preceision_test_gpu` )

When `min_arch == {0, 0}` (i.e. `ShouldIgnorePerformance() == true`), 
* we need to return simply the number of GPUs, and 
* not the number of GPUs with efficient FP16 compute

Because kMinArch is set to `{0, 0}` for ROCm, you cannot use `min_arch == {0, 0}` check as a proxy for `ShouldIgnorePerformance() == true` (like it is done for the CUDA side).  You can get around this in one of two ways
1. Dont think we use kMinArch on the ROCm side at all, so you can let it have the same value as on the CUDA side, which will make it non-zero and the proxy check above, will work as expected
2. Explicitly pass in "ignore_performance" as a boolean flag to `GetNumGPUs`....this is overkill for CUDA, but makes things clear from a ROCm perspective

Reply:
There was another solution. I could call ShouldIgnorePerformance directly in the TENSORFLOW_USE_ROCM conditional. That way I did not need to alter the args for GetNumGPUs. This solved the python: auto_mixed_precision failures locally.